src/Servers/Kestrel/Core/src/Internal/Http3/Http3Stream.cs

```diff
@@ -821,7 +832,45 @@ private static Task ProcessUnknownFrameAsync()
             await OnEndStreamReceived();
         }
 
-        if (!_isMethodConnect && (_parsedPseudoHeaderFields & _mandatoryRequestPseudoHeaderFields) != _mandatoryRequestPseudoHeaderFields)
+        if (_isProtocolWebTransport)
+        {
+            // Requirement in the spec but this is handled by the Http/3 layer already, so leaving commented out for posterity
+            //if (Scheme != SchemeHttps)
+            //{
+            //    throw new Http3StreamErrorException("Scheme must be HTTPS." /*todo unhardcode*/, Http3ErrorCode.MessageError);
+            //}
+
+            if (!_isMethodConnect)
+            {
+                throw new Http3StreamErrorException("Method must be CONNECT." /*todo unhardcode*/, Http3ErrorCode.ProtocolError); // todo should this be a 400 response instead?
```

Comment:
What are your thoughts on this? Chris brought up that it might make sense to throw a 400 error in some of these cases. We settled on the `ProtocolError` because it seemed applicable and more consistent though

Reply:
Related: https://github.com/dotnet/aspnetcore/issues/17861. I don't think we've worked out a good way yet to return 4xx responses from this point in the code.

Reply:
Ok I can look into this when I am working on the Middleware layer and return the 400 from there.